server/src/test/java/io/crate/testing/SqlTransportExecutorTest.java

```diff
@@ -0,0 +1,134 @@
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.testing;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.Random;
+
+import org.junit.Test;
+
+import io.crate.planner.optimizer.Rule;
+import io.crate.planner.optimizer.rule.DeduplicateOrder;
+import io.crate.planner.optimizer.rule.MergeFilterAndCollect;
+import io.crate.planner.optimizer.rule.MoveFilterBeneathGroupBy;
+import io.crate.planner.optimizer.rule.MoveFilterBeneathOrder;
+import io.crate.planner.optimizer.rule.MoveLimitBeneathEval;
+import io.crate.planner.optimizer.rule.MoveOrderBeneathFetchOrEval;
+import io.crate.planner.optimizer.rule.RewriteFilterOnOuterJoinToInnerJoin;
+import io.crate.planner.optimizer.rule.RewriteNestedLoopJoinToHashJoin;
+import io.crate.planner.optimizer.rule.RewriteToQueryThenFetch;
+
+public class SqlTransportExecutorTest {
+
+    @Test
+    public void test_rule_randomization() {
+        List<Class<? extends Rule<?>>> allRules = List.of(
+            DeduplicateOrder.class,
+            MoveFilterBeneathOrder.class,
+            RewriteToQueryThenFetch.class,
+            MoveLimitBeneathEval.class,
+            MergeFilterAndCollect.class,
+            RewriteToQueryThenFetch.class,
+            RewriteNestedLoopJoinToHashJoin.class,
+            RewriteFilterOnOuterJoinToInnerJoin.class,
+            MoveOrderBeneathFetchOrEval.class,
+            MoveFilterBeneathGroupBy.class
+        );
+
+        var result = SQLTransportExecutor.buildRandomizedRuleSessionSettings(new Random(1L),
+                                                                             0.0,
+                                                                             allRules,
+                                                                             List.of());
+        assertThat(result).hasSize(0);
+
+        result = SQLTransportExecutor.buildRandomizedRuleSessionSettings(new Random(1L),
+                                                                         0.3,
+                                                                         allRules,
+                                                                         List.of());
+        assertThat(result).hasSize(3);
```

Comment:
minor, for here and in all other places in the class:
1. you can chain the assertions `.hashSize(3).containsExactly(...)`
2. no need to assert the size since you use assert with containsExactly